public/app/plugins/panel/heatmap-new/fields.ts

```diff
@@ -152,6 +160,26 @@ export const getExemplarsMapping = (heatmapData: HeatmapData, rawData: DataFrame
   return mapping;
 };
 
+const getSparseHeatmapData = (
+  frame: DataFrame,
+  exemplars: DataFrame | undefined,
+  theme: GrafanaTheme2
+): HeatmapData => {
+  if (frame.meta?.type !== DataFrameType.ExperimentalSparseHistogram) {
+    return {
+      warning: 'Expected sparse heatmap format',
+      heatmap: frame,
+    };
+  }
+
```

Comment:
@leeoniya -- I think https://github.com/grafana/grafana/pull/48564 will make this actually work, right?